addons/survey/controllers/main.py

```diff
@@ -585,13 +495,14 @@ def _prepare_result_dict(self, survey, current_filters=None):
     def _get_filter_data(self, post):
         """Returns data used for filtering the result"""
         filters = []
-        for ids in post:
-            #if user add some random data in query URI, ignore it
-            try:
-                row_id, answer_id = ids.split(',')
-                filters.append({'row_id': int(row_id), 'answer_id': int(answer_id)})
-            except:
-                return filters
+        filters_data = post.get('filters')
+        if filters_data:
+            for data in filters_data.split('|'):
```

Comment:
Not your fault, but still looks ugly.
We should refactor this one day with a proper filter structure.